### PR TITLE
Remove kvstore backend deps from datastore package

### PR DESCRIPTION
**Pull request by mrjana**
Currently datastore has dependencies on various kv backends.
This is undesirable if datastore had to be used as a backend
agnostic store management package with it's cache layer. This
PR aims to achieve that.

Signed-off-by: Jana Radhakrishnan <mrjana@docker.com>

### CODE REVIEW

File: Godeps/_workspace/src/github.com/docker/libkv/store/boltdb/boltdb.go [hunk 1]
```diff
@@ -381,7 +379,7 @@ func (b *BoltDB) AtomicPut(key string, value []byte, previous *store.KVPair, opt
 		// doesn't exist in the DB.
 		val = bucket.Get([]byte(key))
 		if previous == nil && len(val) != 0 {
-			return store.ErrKeyModified
+			return store.ErrKeyExists
```
Comment (aboch) [inline Godeps/_workspace/src/github.com/docker/libkv/store/boltdb/boltdb.go]: Shouldn't this be handled in libnetwork as a retry ?

Comment (mrjana) [inline Godeps/_workspace/src/github.com/docker/libkv/store/boltdb/boltdb.go]: Yeah I think this change was made long time back in libkv. But we need to treat both `KeyExists` and `KeyModified` as key modified in datastore. Let me make that change as part of this PR lest we forget.

**Comment by mrjana** [comment]
@aboch Updated datastore to adjust for the error code change in libkv.

**Comment by aboch** [comment]
LGTM

**Pull request merged**